ed25519-program/src/lib.rs

```diff
@@ -58,13 +58,18 @@ pub fn offsets_to_ed25519_instruction(offsets: &[Ed25519SignatureOffsets]) -> In
     }
 }
 
+#[deprecated(since = "2.2.3", note = "Use new_ed25519_instruction_with_signature")]
 pub fn new_ed25519_instruction(keypair: &ed25519_dalek::Keypair, message: &[u8]) -> Instruction {
     let signature = keypair.sign(message).to_bytes();
     let pubkey = keypair.public.to_bytes();
+    new_ed25519_instruction_with_signature(message, &signature, &pubkey)
+}
 
-    assert_eq!(pubkey.len(), PUBKEY_SERIALIZED_SIZE);
-    assert_eq!(signature.len(), SIGNATURE_SERIALIZED_SIZE);
-
+pub fn new_ed25519_instruction_with_signature(
+    message: &[u8],
+    signature: &[u8; SIGNATURE_SERIALIZED_SIZE],
+    pubkey: &[u8; PUBKEY_SERIALIZED_SIZE],
```

Comment:
These changes make a lot of sense.

Reply:
ok great! I tried to follow other verification method signatures like https://docs.rs/ed25519-dalek/latest/ed25519_dalek/fn.verify_batch.html, which go message, then signature, then verifying key